### PR TITLE
Convert move in request handler to use MediatR

### PR DESCRIPTION
**Pull request by kft**
## Description
Convert move in request handler to use MediatR

### CODE REVIEW

File: source/Messaging.Application/Transactions/MoveIn/MoveInRequestHandler.cs [hunk 1]
```diff
@@ -28,7 +30,7 @@
 
 namespace Messaging.Application.Transactions.MoveIn
 {
-    public class MoveInRequestHandler
+    public class MoveInRequestHandler : IRequestHandler<IncomingMessage, Unit>
```
Comment (Korgath) [inline source/Messaging.Application/Transactions/MoveIn/MoveInRequestHandler.cs:33]: nit: "Unit" can be omitted

File: source/Messaging.Application/Transactions/MoveIn/MoveInRequestHandler.cs [hunk 2]
```diff
@@ -91,6 +93,7 @@ public async Task HandleAsync(IncomingMessage incomingMessage)
             transaction.Start(businessProcessResult);
             _moveInTransactionRepository.Add(transaction);
             await _unitOfWork.CommitAsync().ConfigureAwait(false);
```
Comment (Korgath) [inline source/Messaging.Application/Transactions/MoveIn/MoveInRequestHandler.cs:95]: Move out in a unit of work behaviour?

File: source/Messaging.Infrastructure/Configuration/CompositionRoot.cs [hunk 3]
```diff
@@ -202,8 +202,8 @@ public CompositionRoot AddMoveInServices(MoveInConfiguration configuration)
         {
             _services.AddScoped<MoveInNotifications>();
             _services.AddScoped(_ => configuration);
-            _services.AddScoped<MoveInRequestHandler>();
             _services.AddScoped<IMoveInRequester, MoveInRequester>();
+            _services.AddTransient<IRequestHandler<IncomingMessage, Unit>, MoveInRequestHandler>();
```
Comment (Korgath) [inline source/Messaging.Infrastructure/Configuration/CompositionRoot.cs:206]: Maybe we need to rethink how we register handlers.
As of now we have a specific method for register request/notification handlers and now we also register them in other registration services.

The concern is that we loose the overview of registered handlers

**Review by Korgath — CHANGES_REQUESTED** [review]

File: source/Messaging.Application/Transactions/MoveIn/MoveInRequestHandler.cs (hunk 1 above)
Comment (kft) [reply source/Messaging.Application/Transactions/MoveIn/MoveInRequestHandler.cs:33]: But we don't want. I complicates DI registration

File: source/Messaging.Application/Transactions/MoveIn/MoveInRequestHandler.cs (hunk 2 above)
Comment (kft) [reply source/Messaging.Application/Transactions/MoveIn/MoveInRequestHandler.cs:95]: Yes. Next PR :-)

**Review by Korgath — APPROVED** [review]

**Pull request merged**